plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -0,0 +1,657 @@
+	 *
+	 * @return void
+	 */
+	private function sanitize_orderby(): void {
+		// Allowed keys.
+		// TODO: rand, meta keys, etc.
+		$allowed_keys = array( 'ID', 'id', 'type', 'date', 'modified', 'parent' );
+
+		// Translate $orderby to a valid field.
+		$mapping = array(
+			'ID'       => "{$this->tables['orders']}.id",
+			'id'       => "{$this->tables['orders']}.id",
+			'type'     => "{$this->tables['orders']}.type",
+			'date'     => "{$this->tables['orders']}.date_created_gmt",
+			'modified' => "{$this->tables['orders']}.date_updated_gmt",
+			'parent'   => "{$this->tables['orders']}.parent_order_id",
+		);
+
+		$order   = $this->args['order'] ?? '';
+		$orderby = $this->args['orderby'] ?? '';
+
+		if ( 'none' === $orderby ) {
+			return;
+		}
+
+		if ( is_string( $orderby ) ) {
+			$orderby = array( $orderby => $order );
+		}
+
+		$this->args['orderby'] = array();
+		foreach ( $orderby as $order_key => $order ) {
+			if ( isset( $mapping[ $order_key ] ) ) {
+				$this->args['orderby'][ $mapping[ $order_key ] ] = $this->sanitize_order( $order );
+			}
+		}
+	}
+
+	/**
+	 * Makes sure the order in an ORDER BY statement is either 'ASC' o 'DESC'.
+	 *
+	 * @param string $order The unsanitized order.
+	 * @return string The sanitized order.
+	 */
+	private function sanitize_order( string $order ): string {
+		$order = strtoupper( is_string( $order ) ? $order : '' );
+
+		return in_array( $order, array( 'ASC', 'DESC' ), true ) ? $order : 'DESC';
+	}
+
+	/**
+	 * Builds the final SQL query to be run.
+	 *
+	 * @return void
+	 */
+	private function build_query(): void {
+		$this->maybe_remap_args();
+
+		// Build query.
+		$this->process_orders_table_query_args();
+		$this->process_operational_data_table_query_args();
+		$this->process_addresses_table_query_args();
+
+		// Meta queries.
+		if ( ! empty( $this->args['meta_query'] ) ) {
+			$this->meta_query = new OrdersTableMetaQuery( $this );
+
+			$sql = $this->meta_query->get_sql_clauses();
+
+			$this->join  = $sql['join'] ? array_merge( $this->join, $sql['join'] ) : $this->join;
+			$this->where = $sql['where'] ? array_merge( $this->where, array( $sql['where'] ) ) : $this->where;
+
+			if ( $sql['join'] ) {
+				$this->groupby[] = "{$this->tables['orders']}.id";
+			}
+		}
+
+		$this->process_orderby();
+		$this->process_limit();
+
+		$orders_table = $this->tables['orders'];
+
+		// DISTINCT.
+		$distinct = '';
+
+		// SELECT [fields].
+		$this->fields = "{$orders_table}.id";
+		$fields       = $this->fields;
+
+		// SQL_CALC_FOUND_ROWS.
+		if ( ( ! $this->arg_isset( 'no_found_rows' ) || ! $this->args['no_found_rows'] ) && $this->limits ) {
+			$found_rows = 'SQL_CALC_FOUND_ROWS';
+		} else {
+			$found_rows = '';
+		}
+
+		// JOIN.
+		$join = implode( ' ', $this->join );
+
+		// WHERE.
+		$where = '1=1';
+		foreach ( $this->where as $_where ) {
+			$where .= " AND ({$_where})";
+		}
+
+		// ORDER BY.
+		$orderby = $this->orderby ? ( 'ORDER BY ' . $this->orderby ) : '';
```

Comment:
note that `$this->orderby` is declared as `array` in its definition, but it's being cast to `string` here. we can store the generated clause in a local variable (or perhaps in another global variable)

Reply:
Done.